atlasdb-dagger/src/main/java/com/palantir/atlasdb/services/TransactionManagerModule.java

```diff
@@ -80,6 +93,18 @@ public Cleaner provideCleaner(
                 .buildCleaner();
     }
 
+    @Provides
+    @Singleton
+    @Internal
+    public DerivedSnapshotConfig provideDerivedSnapshotConfig(
+            AtlasDbConfig atlasDbConfig, AtlasDbRuntimeConfig atlasDbRuntimeConfig) {
```

Comment:
The @Internal annotation is private, so no other module can depend on the DerivedSnapshotConfig.

This adds a bit of duplication (TestTransactionManagerModule), since TTMM also requires a DerivedSnapshotConfig but is in a different package (so can't just use package private annotation).

If this didn't need to be private, then we could create a provider for `ServiceDiscoveringAtlasSupplier` from ServicesConfig that `provideRawKeyValueService` can use, and `provideDerivedSnapshotConfig` can use the `ServiceDiscoveringAtlasSupplier#getDerivedSnapshotConfig` method rather than what is currently is doing.